pkg/api/types.go

```diff
@@ -3023,6 +3023,10 @@ type NodeList struct {
 type NamespaceSpec struct {
 	// Finalizers is an opaque list of values that must be empty to permanently remove object from storage
 	Finalizers []FinalizerName
+
+	// NetworkPolicy describes the use of NetworkPolicy objects in this namespace. If not specified, then default policies will be applied and NetworkPolicy objects in the Namespace will be ignored
+	// +optional
+	NetworkPolicy *NamespaceNetworkPolicy
```

Comment:
Can we implement this in same way as ResourceQuota?
There is a separate ResourceQuota resource and a ResourceQuota applies to a namespace if it is created in that namespace.
We can similarly add a namespace scoped NetworkPolicy resource.
Wont need to change NamespaceSpec then.